messaging/src/main/java/org/axonframework/messaging/responsetypes/PublisherResponseType.java

```diff
@@ -71,29 +71,28 @@ public Class responseMessagePayloadType() {
     }
 
     @Override
-    public Flux<R> convert(Object queryResponse) {
+    public Publisher<R> convert(Object queryResponse) {
+        if (!projectReactorOnClassPath()) {
+            return null;
+        }
         if (queryResponse == null) {
             return Flux.empty();
-        } else if (Flux.class.isAssignableFrom(queryResponse.getClass())) {
-            return (Flux<R>) queryResponse;
+        } else if (Publisher.class.isAssignableFrom(queryResponse.getClass())) {
+            return Flux.from((Publisher) queryResponse);
         } else if (Iterable.class.isAssignableFrom(queryResponse.getClass())) {
             return Flux.fromIterable((Iterable) queryResponse);
         } else if (Stream.class.isAssignableFrom(queryResponse.getClass())) {
             return Flux.fromStream((Stream) queryResponse);
-        } else if (Mono.class.isAssignableFrom(queryResponse.getClass())) {
```

Comment:
Curious: why is this `Mono` check removed too? Is a `Mono` also a `Publisher`?

Reply:
Exactly!

Reply:
Cool, good to know!